### DISCUSSION TITLE
docker manager build updates

### DISCUSSION THREAD

**Discussion by pyunramura** [renovatebot/renovate#16982] [Request Help]
I'm attempting  to resolve my issue #16826 with a different approach. I've decided to try updating versioned tags again instead of digests since I can have a better overview of major,minor,and patch version bumps and pull in release notes for each new PR. The issue is still regarding linuxserver images, and in my current testing I'm getting hung up on build updates.

The docs state that build updates are handled like patch updates. So in my understanding, even if patch versions would stay the same e.g. `1.1.1.123` to `1.1.1.124`, the build version being updated in the registry would be seen as a patch level update as well.

I think the complication stems from an unorthodox versioning regex I'm using for the bulk of my linuxserver images.

```
^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)((\\.|-)\\w+)?-ls(?<build>\\d+)$
```

Using this regex, I'm attempting to ignore standard build regex `\\.(?<build>\\d+)` corresponding to `1.1.1.123` or the uncommon `1.1.1-a5` in favor of `-ls(?<build>\\d+)` which would correspond to `1.1.1.123-ls123` or `1.1.1-a5-ls123`. I've decided to follow `-ls` builds over "standard" since often the standard build version will not change for months, but the `-ls` build is updated weekly with base image updates.

Using the above regex I'm getting patch and minor version PRs without issue, but build versions are not raising an update.

I've included a few relevant snippets of the logs below

```
DEBUG: found labels in manifest (repository=pyunramura/renovate-test)
       "labels": {
         "build_version": "Linuxserver.io version:- 3.0.8.1507-ls152 Build-date:- 2022-08-04T19:54:47+02:00",
         "maintainer": "aptalca",
         "org.opencontainers.image.authors": "linuxserver.io",
         "org.opencontainers.image.created": "2022-08-04T19:54:47+02:00",
         "org.opencontainers.image.description": "[Sonarr](https://sonarr.tv/) (formerly NZBdrone) is a PVR for usenet and bittorrent users. It can monitor multiple RSS feeds for new episodes of your favorite shows and will grab, sort and rename them. It can also be configured to automatically upgrade the quality of files already downloaded when a better quality format becomes available.  ",
         "org.opencontainers.image.documentation": "https://docs.linuxserver.io/images/docker-sonarr",
         "org.opencontainers.image.licenses": "GPL-3.0-only",
         "org.opencontainers.image.ref.name": "273ec518e0a0695d91ad4d965f84e22770ed309d",
         "org.opencontainers.image.revision": "273ec518e0a0695d91ad4d965f84e22770ed309d",
         "org.opencontainers.image.source": "https://github.com/linuxserver/docker-sonarr",
         "org.opencontainers.image.title": "Sonarr",
         "org.opencontainers.image.url": "https://github.com/linuxserver/docker-sonarr/packages",
         "org.opencontainers.image.vendor": "linuxserver.io",
         "org.opencontainers.image.version": "3.0.8.1507-ls152"
       }
...
DEBUG: packageFiles with updates (repository=pyunramura/renovate-test)
       "config": {
         "docker-compose": [
           {
             "packageFile": "compose/sonarr/docker-compose.yml",
             "deps": [
               {
                 "depName": "ghcr.io/linuxserver/sonarr",
                 "currentValue": "3.0.8.1507-ls147",
                 "replaceString": "ghcr.io/linuxserver/sonarr:3.0.8.1507-ls147",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "versioning": "regex:^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)((\\.|-)\\w+)?(?<build>-ls\\d+)",
                 "sourceUrl": "https://github.com/linuxserver/docker-sonarr",
                 "currentVersion": "3.0.8.1507-ls147",
                 "fixedVersion": "3.0.8.1507-ls147"
               }
             ]
           }
         ]
       }
```

likely something is wrong with my regex, but I'm not really sure where my failure is.

I can create a public example repo if need be, or If more logs would help I can retrieve those as well.

Any insight into this issue would be greatly appreciated.

**Comment by viceice** [comment]
you can't have the same regex group name twice. so you need to use two different configurations

**Reply by pyunramura** [reply]
I'm not sure I understand what you mean. Are you referring to my versioning regex? 

```
^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)((\\.|-)\\w+)?-ls(?<build>\\d+)$
```

I believe I captured each group once. Or are you saying that I can either use patch OR build but not both?

**Reply by viceice** [reply]
ups, saw something wrong. but you would need a fifths version component for that case, which is currently not supported

**Reply by pyunramura** [reply]
I see. My goal with that regex was to ignore the `.1507` portion in favor of the `-ls147` portion for the build group, so I could stay within the four version limit. Would using `((\\.|-)\\w+)?` let me work around the fourth version limit? Do I need five version components if I don't capture `.1507` in a group?

Here is what I'm trying to achieve with each regex group in a more digestible format if that helps at all.

|major|minor|patch|ignore|build|
|---|---|---|---|---|
|3|.0|.8|.1507|-ls147|
| `(?<major>\\d+)` | `\\.(?<minor>\\d+)` | `\\.(?<patch>\\d+)` | `((\\.\|-)\\w+)?` | `-ls(?<build>\\d+)` |

I sincerely appreciate your help with this issue
